### PR TITLE
Make UserMenu match DSM

### PR DESCRIPTION
**Pull request by daileytj**
Fixes PXBLUE-1899.

#### Changes proposed in this Pull Request:
- Update InfoListItem typography variants to match DSM
- Make UserMenu match DSM

### CODE REVIEW

**Review by joebochill — COMMENTED** [review]
What about the font colors for the icons and the text, are those matching the DSM?

**Comment by daileytj** [comment]
> What about the font colors for the icons and the text, are those matching the DSM?

Good catch! I was so focused on typography stuff that I didn't even question the colors. I'll update those to match.

**Comment by daileytj** [comment]
Simulator Screen Shot - iPhone 11 - 2021-03-02 at 07 56 39

File: components/src/core/user-menu/user-menu.tsx [hunk 1]
```diff
@@ -86,6 +96,7 @@ export const UserMenu: React.FC<UserMenuProps> = (props) => {
                             iconColor={iconColor || menuItem.iconColor}
                             fontColor={fontColor || menuItem.fontColor}
                             backgroundColor={backgroundColor || menuItem.backgroundColor}
+                            styles={{ title: { fontSize: 16, fontWeight: '400' } }}
```
Comment (joebochill) [inline components/src/core/user-menu/user-menu.tsx:99]: I believe specifying the numerical fontWeight doesn't work as expected on one of the OSs. Safer way to do this is to pull the font family/weight from the theme (https://github.com/pxblue/react-native-component-library/blob/dev/components/src/core/list-item-tag/list-item-tag.tsx#L44):
```suggestion
                            styles={{ title: { 
                                fontSize: 16, 
                                fontFamily: theme.fonts.regular.fontFamily,
                                fontWeight: theme.fonts.regular.fontFamily
                            }}}
```

**Comment by joebochill** [comment]
Changes look good, I'm just trying to verify the colors in your screenshot. The icon is gray[500] and the text is black[500] right?

File: components/src/core/user-menu/user-menu.tsx [hunk 2]
```diff
@@ -86,6 +96,13 @@ export const UserMenu: React.FC<UserMenuProps> = (props) => {
                             iconColor={iconColor || menuItem.iconColor}
                             fontColor={fontColor || menuItem.fontColor}
                             backgroundColor={backgroundColor || menuItem.backgroundColor}
+                            styles={{
+                                title: {
+                                    fontSize: 16,
+                                    fontFamily: theme.fonts.regular.fontFamily,
+                                    fontWeight: theme.fonts.regular.fontWeight,
+                                },
+                            }}
```
Comment (joebochill) [inline components/src/core/user-menu/user-menu.tsx:105]: Colors look good, but adding this styles prop breaks the end-user style overrides functionality. You need to merge these style with any overrides that are being passed in through the menuItem, e.g.:

```
                {menuItems &&
                    menuItems.map((menuItem: InfoListItemProps, index: number) => {
                        const menuStyles = menuItem.styles || {};
                        return (
                            <InfoListItem
                                {...menuItem}
                                key={index}
                                onPress={(): void => {
                                    closeMenu();
                                    if (menuItem.onPress) menuItem.onPress();
                                }}
                                iconColor={iconColor || menuItem.iconColor}
                                fontColor={fontColor || menuItem.fontColor}
                                backgroundColor={backgroundColor || menuItem.backgroundColor}
                                styles={
                                    Object.assign(menuStyles, {
                                        title: Object.assign({
                                            fontSize: 16,
                                            fontFamily: theme.fonts.regular.fontFamily,
                                            fontWeight: theme.fonts.regular.fontWeight,
                                        }, menuStyles.title)
                                    })
                                }
                            />
                        )
                    })}
```

**Review by joebochill — CHANGES_REQUESTED** [review]

**Comment by daileytj** [comment]
> Changes look good, I'm just trying to verify the colors in your screenshot. The icon is gray[500] and the text is black[500] right?

Correct.

**Review by joebochill — APPROVED** [review]

**Pull request merged**